kendzi3d-jogl-model-render/src/main/java/kendzi/jogl/model/render/ModelRender.java

```diff
@@ -142,53 +160,54 @@ private void draw(GL2 gl, Model model) {
                     gl.glEnd();
                 }
 
-                unsetupTextures(gl, material, useTextures);
-
+                if (drawTextures) {
+                    if (model.useTextureAlpha)
+                        disableTransparentText(gl);
+                    unsetupTextures(gl, material, mesh.hasTexture);
+                }
             }
 
             gl.glColor3f(1.0f, 1.0f, 1.0f);
 
         } catch (RuntimeException e) {
             throw new RuntimeException("error model: " + model.getSource() + " mesh: " + mi + " ("
                     + (model.mesh[mi] != null ? model.mesh[mi].name : "") + ")" + " face: " + fi, e);
-        }
+        } finally {
 
```

Comment:
The finally clause discussed previously moved here.

It disables GL_CULL_FACE and resets GL_LIGHT_MODEL_TWO_SIDE  in any case before leaving the draw method of ModelRender.

Reply:
That runtime exceptions, like array index out of range were huge problem in past, they were caused by wrong models. Exception was added to help locate them. If you don't produce faulty model you wont see that error. Still sometimes because some numerical errors or by adding new feature they may appear in future. Yes, i known that lot thinks should be covered by tests...